yamllint/rules/key_ordering.py

```diff
@@ -86,6 +134,8 @@ def check(conf, token, prev, next, nextnext, context):
     if 'stack' not in context:
         context['stack'] = []
 
+    setlocale(LC_COLLATE, conf['locale'])
```

Comment:
With this, running `yamllint` on a directory with 4 or 5 YAML files calls `setlocale(LC_COLLATE, ...)` 657 times. This is a lot.

Moreover, calling `setlocale()` could theoretically affect other parts of yamllint.

→ Is there a way to call an equivalent of `strcoll()` by passing the `conf['locale']` directly? (I quickly looked at [the `locale` docs](https://docs.python.org/3/library/locale.html#locale.setlocale), haven't found.)
→ If calling `setlocale()` is really needed, can we do it only once?

Reply:
657 times is a lot - I hope you did not count that ;)

We can't avoid calling `setlocale()`, because `strcoll()` only works with that. The only other way I found to do that kind of sorting would be to use `PyICU`. But adding an external dependency with c++ libs as dependencies just for that part seems like overkill to me.
 
In theory this setting could affect other parts of yamllint, but the docs say:

>locale.LC_COLLATE
>   Locale category for sorting strings. The functions strcoll() and strxfrm() of the locale module are affected.

If only those two functions rely on this setting - and they are not used anywhere else (they are not right now), then nothing should be affected.

Calling setlocale only once seems plausible. I have not digged in deeper in yamllints source code, but just copy-pasted my way through so far ;), so maybe you can give me a hint: I guess there is some place where we can set global options instead of rule-specific?

We would just need to make `locale` global and call it only when reading the config file. Which seems better in regard to the "what if we use strcoll somewhere else as well - will that not affect us there as well" problem.